navigator/backend/src/main/scala/com/digitalasset/navigator/model/Ledger.scala

```diff
@@ -126,18 +127,24 @@ case class Ledger(
     }
 
   private def withContractCreatedInEvent(contract: Contract, event: ContractCreated): Ledger = {
+    val isStakeHolder = contract.signatories.contains(forParty) || contract.observers.contains(
+      forParty)
     if (useDatabase) {
-      db.insertContract(contract)
+      if (isStakeHolder) db.insertContract(contract)
```

Comment:
Minor, not a necessary change: since the behavior is so different it could make sense to have to helpers that are invoked based on the value of `isStakeholder`.

Reply:
Related: Long term, we should remove the in-memory store (along with all custom query functionality) and only keep the database.